pydantic_ai_slim/pydantic_ai/messages.py

```diff
@@ -1024,6 +1024,9 @@ class TextPart:
     part_kind: Literal['text'] = 'text'
     """Part type identifier, this is available on all parts as a discriminator."""
 
+    provider_details: dict[str, Any] | None = None
```

Comment:
For now, let's only add this to fields that have a `provider_name`, meaning `ThinkingPart`, `BuiltinTool{Call,Return}Parts`, and `FilePart`, as those are the ones where we've previously found it necessary to branch behavior based on which provider they came from, in order to interpret the `id` or `args` in a special way, for example.

Since this field is only meant to be written to by model classes, we can introduce it on new parts as there is reason to.